tests/Unit/Evolution/DiscontinuousGalerkin/SlopeLimiters/Test_Weno.cpp

```diff
@@ -0,0 +1,878 @@
+}
+
+template <size_t VolumeDim>
+using VariablesMap = std::unordered_map<
+    std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>,
+    Variables<tmpl::list<ScalarTag, VectorTag<VolumeDim>>>,
+    boost::hash<std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>>>;
+
+template <size_t VolumeDim>
+std::unordered_map<
+    std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>,
+    typename SlopeLimiters::Weno<
+        VolumeDim, tmpl::list<ScalarTag, VectorTag<VolumeDim>>>::PackagedData,
+    boost::hash<std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>>>
+make_neighbor_data_from_neighbor_vars(
+    const Element<VolumeDim>& element, const Mesh<VolumeDim>& mesh,
+    const std::array<double, VolumeDim>& element_size,
+    const VariablesMap<VolumeDim>& neighbor_vars) noexcept {
+  const auto make_tuple_of_means = [&mesh](
+      const Variables<tmpl::list<ScalarTag, VectorTag<VolumeDim>>>&
+          vars_to_average) noexcept {
+    tuples::TaggedTuple<::Tags::Mean<ScalarTag>,
+                        ::Tags::Mean<VectorTag<VolumeDim>>>
+        result;
+    get(get<::Tags::Mean<ScalarTag>>(result)) =
+        mean_value(get(get<ScalarTag>(vars_to_average)), mesh);
+    for (size_t d = 0; d < VolumeDim; ++d) {
+      mean_value(get<VectorTag<VolumeDim>>(vars_to_average).get(d), mesh);
+    }
+    return result;
+  };
+
+  std::unordered_map<
+      std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>,
+      typename SlopeLimiters::Weno<
+          VolumeDim, tmpl::list<ScalarTag, VectorTag<VolumeDim>>>::PackagedData,
+      boost::hash<std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>>>
+      neighbor_data{};
+
+  for (const auto& neighbor : element.neighbors()) {
+    const auto dir = neighbor.first;
+    const auto id = *(neighbor.second.cbegin());
+    const auto dir_and_id = std::make_pair(dir, id);
+    neighbor_data[dir_and_id].volume_data = neighbor_vars.at(dir_and_id);
+    neighbor_data[dir_and_id].means =
+        make_tuple_of_means(neighbor_vars.at(dir_and_id));
+    neighbor_data[dir_and_id].mesh = mesh;
+    neighbor_data[dir_and_id].element_size = element_size;
+  }
+
+  return neighbor_data;
+}
+
+template <size_t VolumeDim>
+void test_weno_work(
+    const SlopeLimiters::WenoType& weno_type, const Element<VolumeDim>& element,
+    const Mesh<VolumeDim>& mesh,
+    const std::array<double, VolumeDim>& element_size,
+    const Variables<tmpl::list<ScalarTag, VectorTag<VolumeDim>>>& local_vars,
+    const std::unordered_map<
+        std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>,
+        typename SlopeLimiters::Weno<
+            VolumeDim,
+            tmpl::list<ScalarTag, VectorTag<VolumeDim>>>::PackagedData,
+        boost::hash<std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>>>&
+        neighbor_data,
+    const VariablesMap<VolumeDim>& expected_neighbor_modified_vars,
+    // NOLINTNEXTLINE(google-runtime-references):
+    Approx& local_approx = approx) noexcept {
```

Comment:
It's not clear to me why this needs to be a mutable reference. Could you explain?

Reply:
The approx doesn't want to be const. I found some other instances in the code that take it by mutable reference so I went with that... but it could also be passed by value.

Reply:
Awe :( Taking by value seems somewhat cleaner so we don't need the `NOLINT` but :man_shrugging: